### DISCUSSION TITLE
How to audit errors-in-console assertion ignoring CSP errors

### DISCUSSION THREAD

**Discussion by 462702985** [GoogleChrome/lighthouse#15195] [Q&A]
I want to ignore content security policy error in console, and I found there is an option to ignore these errors: https://github.com/GoogleChrome/lighthouse/pull/9480/files
How can I add this ignore pattern option to my assert matrix so errors-in-console assertion can pass? For example:
matrix: {
    'matchingUrlPattern': '.*',
    'assertions': {
      'errors-in-console': ['error', {'minScore': 1}],
    }
  }

**Comment by adamraine** [comment]
I'm not sure what this assert matrix is. In Lighthouse if you want to adjust audit specific options you need to do so on the [config](https://github.com/GoogleChrome/lighthouse/blob/main/docs/configuration.md):

```js
const config = {
  extends: 'lighthouse:default',
  audits: [
    {path: 'errors-in-console', options: {ignoredPatterns: ['pattern1']}}
  ]
}
```

**Reply by 462702985** [reply]
There are many configs, I tried to add this in all places but still shows assertion error. There are some pictures might help locating the problem.
![image](https://github.com/GoogleChrome/lighthouse/assets/85102180/3e4e3092-b442-447a-a2cd-e57e1ec41b19)
![image](https://github.com/GoogleChrome/lighthouse/assets/85102180/fcda76dd-0894-4e17-a004-9a8aaca365d0)
![image](https://github.com/GoogleChrome/lighthouse/assets/85102180/7fbb8ed5-be59-43f8-a40b-7760a14285f1)

**Comment by connorjclark** [comment]
You need to use the following Lighthouse config:

```js
export default {
  extends: 'lighthouse:default',
  audits: [
    {path: 'errors-in-console', options: {
      ignoredPatterns: '...',
    }},
  ],
};
```



This is how you configure the Lighthouse configuration used in LHCI: https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/configuration.md#custom-lighthouse-config

```
{
  "ci": {
    "collect": {
      "settings": {
        "configPath": "./path/to/lighthouse/config.js"
      }
    }
  }
}
```

**Reply by 462702985** [reply]
I added lighthouse config, but I don't know where I can add LHCI config path. Should I change cli.js?

![image](https://github.com/GoogleChrome/lighthouse/assets/85102180/088dd122-f116-4ccc-8992-1776fceeb612)
